File: src/pathfinder_combat_simulator/core/attack_result.py
from typing import List

class AttackResult:
    """Represents the result of an attack roll"""
    def __init__(self, attacker: str, target: str, attack_name: str):
        self.attacker = attacker
        self.target = target
        self.attack_name = attack_name
        self.attack_roll = 0
        self.total_attack_bonus = 0
        self.target_ac = 0
        self.is_hit = False
        self.is_critical_threat = False
        self.is_critical_hit = False
        self.damage_rolls: List[int] = [] # Changed from list to List[int] for clarity
        self.total_damage = 0
        self.damage_taken = 0
        self.special_effects: List[str] = [] # Changed from list to List[str] for clarity

File: src/core/attackResult.ts
// Forward declaration for Combatant if needed, or use a simpler type like string/object for now
// import { Combatant } from "./combatant";

export class AttackResult {
  attacker_name: string; // Changed from attacker to attacker_name to avoid conflict if using Combatant type
  target_name: string;   // Changed from target to target_name
  attack_name: string;
  attack_roll: number;
  total_attack_bonus: number;
  target_ac: number;
  is_hit: boolean;
  is_critical_threat: boolean;
  is_critical_hit: boolean;
  damage_rolls: number[];
  total_damage: number;
  damage_taken: number; // Damage actually dealt after DR, resistances etc.
  special_effects: string[];

  constructor(attacker_name: string, target_name: string, attack_name: string) {
    this.attacker_name = attacker_name;
    this.target_name = target_name;
    this.attack_name = attack_name;
    this.attack_roll = 0;
    this.total_attack_bonus = 0;
    this.target_ac = 0;
    this.is_hit = false;
    this.is_critical_threat = false;
    this.is_critical_hit = false;
    this.damage_rolls = [];
    this.total_damage = 0;
    this.damage_taken = 0;
    this.special_effects = [];
  }
}
